src/xrpld/app/tx/detail/MPTokenAuthorize.cpp

```diff
@@ -83,6 +83,18 @@ MPTokenAuthorize::preclaim(PreclaimContext const& ctx)
                 return tecHAS_OBLIGATIONS;
             }
 
+            if (ctx.view.rules().enabled(featureTokenEscrow) &&
+                (*sleMpt).isFieldPresent(sfEscrowedAmount) &&
+                (*sleMpt)[~sfEscrowedAmount] != 0)
```

Comment:
It doesn't really need the amendment check. Also can be simplified to 
```suggestion
            if ( (*sleMpt)[~sfEscrowedAmount].value_or(0) != 0)
```